src/huggingface_hub/inference_api.py

```diff
@@ -0,0 +1,108 @@
+import json
+import logging
+from typing import Optional
+
+import requests
+
+from .hf_api import HfApi, HfFolder
+from .validation import ALL_TASKS, check_inputs, check_params
+
+
+logger = logging.getLogger(__name__)
+
+
+ENDPOINT = "https://api-inference.huggingface.co/pipeline"
+
+
+class InferenceApi:
+    """Client to configure requests and make calls to the HuggingFace Inference API.
+
+    Example:
+
+            >>> from huggingface_hub.inference_api import InferenceApi
+
+            >>> # Mask-fill example
+            >>> api = InferenceApi("bert-base-uncased")
+            >>> api.set_inputs(inputs="The goal of life is [MASK].")
+            >>> api.call()
+
+            >>> # Question Answering example
+            >>> api = InferenceApi("deepset/roberta-base-squad2")
+            >>> api.set_inputs(question="What's my name?", context="My name is Clara and I live in Berkeley.")
+            >>> api.call()
+
+            >>> # Zero-shot example
+            >>> api = InferenceApi("typeform/distilbert-base-uncased-mnli")
+            >>> api.set_inputs(inputs="Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!")
+            >>> api.set_params(candidate_labels=["refund", "legal", "faq"])
+            >>> api.call()
+    """
+
+    def __init__(
+        self, repoId: str, task: Optional[str] = None, gpu: Optional[bool] = False
+    ):
+        """Inits InferenceApi headers and API call information.
+
+        Args:
+            repoId (``str``): Id of model (e.g. `bert-base-uncased`).
+            task (``str``, `optional`, defaults ``None``): Whether to force a task instead of using task specified in repository.
+            gpu (``bool``, `optional`, defaults ``None``): Whether to use GPU instead of CPU for inference(requires Startup plan at least).
+        """
+        token = HfFolder.get_token()
```

Comment:
Careful this token is NOT the api token.
I don't think there is a programmatic way to retrieve this api token.

Reply:
I think you're right at the moment. But we can add a way to get it if needed

Reply:
q: Hmm so when I was doing requests with the current, no user was being accounted for the requests?

Would you then suggest to ask users to specify the wanted API token, which could actually be from their organization? Or would you suggest to retrieve user API token, and let users specify their organization token instead?

Reply:
No it's better if it's transparent, it's just that this does not use the `apiToken` but the other `token`. It's not really meant for the API but it does work.